test/ConfigDistributor.spec.js

```diff
@@ -1,198 +1,121 @@
+    it('determines the correct data to distribute to the JavaScript layer', () => {
+      const expectedData = JSON.stringify(cannedData.expectedOutput.js);
 
-      it('returns a promise that will be rejected', () => {
-        const invalidDirName = (function () {
-          const parts = new Array(256);
-          parts.fill('a');
-          return parts.join('');
-        }());
-        return expect(
-          ConfigDistributor.writeDirectory(invalidDirName)
-        ).to.be.rejected;
-      });
+      filesystemMock.expects('writeFile').once().withArgs(expectedData);
 
+      return distributor.distribute(consolidatorFixtures.forJsOnly)
+        .then(() => {
+          filesystemMock.verify();
+        });
     });
 
-    context('when there no error', () => {
-
-      let validDirName;
+    it('determines the correct data to distribute to the Sass layer', () => {
+      const expectedData = cannedData.expectedOutput.sass;
 
-      beforeEach(() => {
-        validDirName = path.join(__dirname, '/tmp/');
-        if (fs.existsSync(validDirName)) {
-          fs.rmdirSync(validDirName);
-        }
-        expect(fs.existsSync(validDirName)).to.be.false;
-      });
+      filesystemMock.expects('writeFile').once().withArgs(expectedData);
 
-      afterEach(() => {
-        fs.rmdirSync(validDirName);
-      });
-
-      it('returns a promise that will not be rejected', () => {
-        return expect(
-          ConfigDistributor.writeDirectory(validDirName)
-        ).to.not.be.rejected;
-      });
-
-      it('writes the directory', () => {
-        return ConfigDistributor.writeDirectory(validDirName).then(() => {
-          return expect(fs.existsSync(validDirName)).to.be.true;
+      return distributor.distribute(consolidatorFixtures.forSassOnly)
+        .then(() => {
+          filesystemMock.verify();
         });
-      });
 
     });
 
   });
 
-  describe('writeFile static method', () => {
-
-    let filepath;
-    let data;
-
-    beforeEach(() => {
-      data = standAloneConfigFixture;
-      filepath = '/tempDir/fileFile.tmp';
-    });
-
-    context('when there is an error writing the file', () => {
+  context('when writing files', () => {
 
-      let fileWriterRejectionMock;
-
-      beforeEach(() => {
-        fileWriterRejectionMock = () => {
-          return Promise.reject();
-        }
-      });
+    it('attempts to distribute the JavaScript layer to the correct path', () => {
+      const cannedConfigToWrite = JSON.stringify(cannedData.expectedOutput.js);
+      const expectedDirectory = paths.output.jsonFile.directory;
+      const expectedFilename = paths.output.jsonFile.filename;
 
-      it('returns a promise that will be rejected', () => {
-        return expect(
-          ConfigDistributor.writeFile(data, filepath, fileWriterRejectionMock)
-        ).to.be.rejected;
-      });
+      filesystemMock.expects('writeFile').once().withArgs(cannedConfigToWrite, expectedDirectory, expectedFilename);
```

Comment:
the expectation seem very readable and the contract easier to understand (a single call, so far)